### ISSUE TITLE
Poisson logpdf inconsistency

### ISSUE THREAD

**Issue by erathorn** [JuliaStats/Distributions.jl#1189]
The evaluation of the `logpdf` on the Poisson distribution gives inconsistent results.

```
my_poisson = Poisson(0.1)

# case 1
# this returns equal results
logpdf(my_poisson, 1) # -2.402585092994046
logpdf(my_poisson, 1.0) # -2.402585092994046

# case 2
# results approximately equal
logpdf(my_poisson, 121) # -741.3209747791544
logpdf(my_poisson, 121.0) # -741.3045777054521

# case 3
# !!!!unequal results!!!!
logpdf(my_poisson, 122) # -748.4275809168819
logpdf(my_poisson, 122.0) # -Inf
```

The expected result would be to get the same result in the respective function evaluations in the different cases. 

I am running

- Julia 1.4.2
- Distributions v0.23.12

**Comment by andreasnoack** [comment]
This one is actually rather unfortunate. Because of https://github.com/JuliaStats/Distributions.jl/blob/d6de7214e6d84df77d1eb3ec7a98d23862d1232d/src/univariates.jl#L541 we are calling the `logpdf` implementation in Rmath only for integer inputs. The `Float64` method ends up calling https://github.com/JuliaStats/Distributions.jl/blob/d6de7214e6d84df77d1eb3ec7a98d23862d1232d/src/univariates.jl#L342. I think we might simply want to remove the ternary and make all the definitions in https://github.com/JuliaStats/Distributions.jl/blob/d6de7214e6d84df77d1eb3ec7a98d23862d1232d/src/univariates.jl#L558-L571 for `Real`.

**Comment by devmotion** [comment]
Note that this issue is fixed by https://github.com/JuliaStats/Distributions.jl/pull/1171 (though @andreasnoack's suggestion might still be useful for other similar issues). With the linked PR I get
```julia
julia> using Distributions

julia> my_poisson = Poisson(0.1)
Poisson{Float64}(λ=0.1)

julia> logpdf(my_poisson, 1)
-2.402585092994046

julia> logpdf(my_poisson, 1.0)
-2.402585092994046

julia> logpdf(my_poisson, 121)
-741.3209747791544

julia> logpdf(my_poisson, 121.0)
-741.3209747791544

julia> logpdf(my_poisson, 122)
-748.4275809168819

julia> logpdf(my_poisson, 122.0)
-748.4275809168819
```
i.e., the results are exactly the same in all cases.

**Comment by andreasnoack** [comment]
Closed by #1171
